text/0000-let.md

```diff
@@ -0,0 +1,265 @@
+- Start Date: 2017-01-14
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Introduce a `let` keyword to bind values to names within a handlebars
+template.
+
+Block form:
+
+```handlebars
+{{#let "Milk" "Cereal" as |one two|}}
+  Breakfast:
+  1. {{one}}
+  2. {{two}}
+{{/let}}
+```
+
+or inline:
+
+```handlebars
+{{let one="Milk" two="Cereal"}}
+
+Breakfast:
+1. {{one}}
+2. {{two}}
+```
+
+# Motivation
+
+The primary method for supplying named values to a template is via
+properties on its backing scope (either a controller or a
+component). These properties are specified in JavaScript which, while
+as a mechanism is very well understood by the community, also comes
+with several drawbacks.
+
+The first is that it spreads the requisite knowledge to understanding
+a template across two places. In other words, it is impossible to understand
+what exactly it does without simultaneously holding both the `.hbs`
+and `.js` files together and continuously merging them with your
+mind. The lack of this overhead is one of the commonly cited strengths
+of JSX over Handlebars.
```

Comment:
I've always considered this to be a strength of Ember - not mixing invocation and implementation both in the template layer but keeping them apart. I generally think of the template context as an API for the template. When you use an API you don't want to deal with all of its internals but rather use it by invoking properly named properties and actions and assume it does the right thing under the hood. That takes a lot of complexity out of the template layer which is actually a good thing IMO.

I also gave a talk about the topic at last year's Emberfest: https://speakerdeck.com/marcoow/templates-and-logic-in-ember - hope videos will be up soon.